packages/expo/android/src/main/java/expo/modules/ReactActivityDelegateWrapper.kt

```diff
@@ -225,7 +225,7 @@ class ReactActivityDelegateWrapper(
      *
      * TODO (@bbarthec): fix it upstream?
      */
-    if (delegate.reactInstanceManager.currentReactContext == null) {
+    if (!BuildConfig.IS_NEW_ARCHITECTURE_ENABLED && delegate.reactInstanceManager.currentReactContext == null) {
```

Comment:
```suggestion
    if (!ReactFeatureFlags.enableBridgelessArchitecture && delegate.reactInstanceManager.currentReactContext == null) {
```

that's more precise for bridgeless.
it's a super edge case. are you sure bridgeless does't have the similar problem? we could test the edge case later.

Reply:
I'd have to try and reproduce on bridgeless but for now, this fixes a lot of issues we had in bridgeless testing. We can land it and try to reproduce during the next round of QA. What do you think?